lib/schema/builders/json-schema-builder.ts

```diff
@@ -56,6 +56,12 @@ export class JsonSchemaBuilder<TEntity extends Entity> extends SchemaBuilder<TEn
           ...this.buildWeight(fieldDef),
           ...this.buildIndexed(fieldDef),
         ]
+      case 'binary':
+        if (fieldDef.vector)
+          console.warn(`You have marked the ${fieldDef.type} field '${field}' as vector but RediSearch doesn't support the VECTOR argument on a field for JSON. Ignored.`);
```

Comment:
I'm pretty sure it is supported

Reply:
I've re-enabled it, but it doesn't work when I tried it (against the latest `redis-stack-server` image on docker hub, and with a numeric array for the field). I based the "not supported" on the comment on Redis Insight:

<img width="295" alt="Screen Shot" src="https://user-images.githubusercontent.com/304910/177898577-75a8e1a4-1644-43de-92a9-db680be6c920.png">

<img width="295" alt="Screen Shot 1" src="https://user-images.githubusercontent.com/304910/177899154-d5244026-4a5b-4179-84f5-6f2add26516b.png">

Maybe it's soon-to-be-released as there are JSON + Vector commits for RediSearch ...

Reply:
I haven't been able to find any definitive docs on what format the JSON vector is expecting - maybe it isn't the encoded bytes but instead the array of floats? If so, the binary field should probably become vector instead so the same value can be set regardless of it being HASH or JSON, or the lib should convert it to a typed array of floats.

Reply:
Confirmed via experimentation - it's expecting the array of floats, not bytes

Reply:
Had a few more attempts, but so far haven't been able to get vector similarity working with JSON. Tried array of bytes, array of floats, all result in index failures (using latest redis-stack-server image 7.0.0-RC6)

Reply:
@CaptainCodeman VSS doesn't work with JSON yet. This is a feature we are working on! Bear with us

Reply:
Thanks!, that's what I thought / deduced - any discussion about the format to use is maybe moot for now then :)

Reply:
@CaptainCodeman, we have basic JSON support on the `edge` build of RediSearch. This is the `master` branch build. You can see a simple test in https://github.com/RediSearch/RediSearch/pull/2704/files in `test_vecsim.py` (or in `test_vecsim.py` if you pull RediSearch `master` branch) 
I want to make sure this PR will be future-proof with our support for VSS on JSON

Reply:
Thanks, I'll wait for it to be available in one of the docker images before picking this up again, but I think it will make sense for the field type to be a typed array (`Float32Array`) rather than a buffer.

Reply:
Looks like this may be out now, so I'll revisit this PR ...